src/tokenizer.ts

```diff
@@ -1485,12 +1480,50 @@ export class Tokenizer extends DiagnosticEmitter {
         ) {
           ++this.pos;
         }
-        while (this.pos < end && isDecimalDigit(text.charCodeAt(this.pos))) {
-          ++this.pos;
+        this.readDecimalFloatPartial();
+      }
+    }
+    let floatString = text.substring(start, this.pos).replaceAll("_", "");
+    return parseFloat(floatString);
```

Comment:
```suggestion
    return parseFloat(text.substring(start, this.pos).replaceAll("_", ""));
```

Reply:
Perhaps [this](https://github.com/AssemblyScript/assemblyscript/pull/1385/files#r453169401) even better